### PR TITLE
Add filter to users and groups endpoints

### PR DESCRIPTION
**Pull request by mdellweg**
fixes #7975
https://pulp.plan.io/issues/7975

### CODE REVIEW

**Review by daviddavis — APPROVED** [review]
LGTM. A test would be nice but I leave that at your discretion.

**Comment by mdellweg** [comment]
Writing a test for the user filters seems to be hard, as long as you cannot manipulate them though the rest API.

File: pulpcore/app/viewsets/user.py [hunk 1]
```diff
@@ -23,6 +25,23 @@
...
+class UserFilter(BaseFilterSet):
+    """
+    FilterSet for User.
+    """
+
+    class Meta:
+        model = User
+        fields = {
+            "username": ["exact", "in", "contains"],
+            "first_name": ["exact", "in", "contains"],
```
Comment (ipanova) [inline pulpcore/app/viewsets/user.py:37]: i suggest adding 'icontains' for case insensitive match on the fileds where it makes sense

**Review by ipanova — APPROVED** [review]
a suggestion but lgtm in an ycase https://github.com/pulp/pulpcore/pull/1056/files#r542696899

File: pulpcore/app/viewsets/user.py [hunk 2]
```diff
@@ -32,6 +51,8 @@ class UserViewSet(NamedModelViewSet, mixins.RetrieveModelMixin, mixins.ListModel
     """
 
     endpoint_name = "users"
+    filterset_class = UserFilter
+    filter_backends = (OrderingFilter, DjangoFilterBackend)
```
Comment (ipanova) [inline pulpcore/app/viewsets/user.py:55]: will you also specify ordering?

Comment (mdellweg) [reply pulpcore/app/viewsets/user.py:55]: I added some tests too.

**Review by ipanova — APPROVED** [review]

**Pull request merged**